web/packages/teleport/src/JoinTokens/JoinTokens.tsx

```diff
@@ -0,0 +1,398 @@
+          attempt={deleteTokenAttempt}
+        />
+      )}
+      {resources.status === 'editing' && (
+        <ResourceEditor
+          docsURL="https://goteleport.com/docs/reference/join-methods"
+          title={'Edit token'}
+          text={resources.item.content}
+          name={resources.item.name}
+          isNew={false} // only editting is allowed
+          onSave={handleSave}
+          onClose={resources.disregard}
+          directions={<Directions />}
+          kind={'join_token'}
+        />
+      )}
+    </FeatureBox>
+  );
+};
+
+export function searchMatcher(
+  targetValue: any,
+  searchValue: string,
+  propName: keyof JoinToken & string
+) {
+  if (propName === 'roles') {
+    return targetValue.some((role: string) =>
+      role.toUpperCase().includes(searchValue)
+    );
+  }
+}
+
+const renderRolesCell = ({ roles }: JoinToken) => {
+  return (
+    <Cell>
+      {roles.map(role => (
+        <StyledLabel key={role}>{role}</StyledLabel>
+      ))}
+    </Cell>
+  );
+};
+
+const NameCell = ({ token }: { token: JoinToken }) => {
+  const { id, safeName, method } = token;
+  const [hovered, setHovered] = useState(false);
+  return (
+    <Cell
+      align="left"
+      style={{
+        minWidth: '320px',
+        fontFamily: 'monospace',
+        whiteSpace: 'nowrap',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+      }}
+      onMouseEnter={() => setHovered(true)}
+      onMouseLeave={() => setHovered(false)}
+    >
+      <Flex alignItems="center" gap={2}>
+        <Text
+          css={`
+            text-overflow: clip;
+            overflow-x: auto;
+          `}
+        >
+          {method !== 'token' ? id : safeName}
+        </Text>
+        {hovered && <CopyButton name={id} />}
+      </Flex>
+    </Cell>
+  );
+};
+
+const StyledLabel = styled(Label)`
+  height: 20px;
+  margin: 1px 0;
+  margin-right: ${props => props.theme.space[2]}px;
+  background-color: ${props => props.theme.colors.interactive.tonal.neutral[0]};
+  color: ${props => props.theme.colors.text.main};
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  line-height: 20px;
+`;
+
+function TokenDelete({
+  token,
+  onDelete,
+  onClose,
+  attempt,
+}: {
+  token: JoinToken;
+  onDelete: (token: string) => Promise<any>;
+  onClose: () => void;
+  attempt: Attempt<void>;
+}) {
+  return (
+    <Dialog
+      dialogCss={() => ({ maxWidth: '500px', width: '100%' })}
+      disableEscapeKeyDown={false}
+      onClose={close}
+      open={true}
+    >
+      <DialogHeader>
+        <DialogTitle>Delete Join Token?</DialogTitle>
+      </DialogHeader>
+      <DialogContent>
+        {attempt.status === 'error' && <Alert children={attempt.statusText} />}
+        <Text mb={4}>
+          You are about to delete join token
+          <Text bold as="span">
+            {` ${token.safeName}`}
+          </Text>
+          . This will not remove any resource that used this join token to join
```

Comment:
Should this be "resources"?

Reply:
I think we should reword "join token to join" too

Reply:
Hmm, I think this copy should change quite a bit. I'm not sure it's 100% accurate anymore because it also depends on token type (I believe this is only true with join method type token)

I'll revisit this entire blurb